pkg/controller/network_policy.go

```diff
@@ -501,6 +534,35 @@ func (c *Controller) fetchSelectedPorts(namespace string, selector *metav1.Label
 	return ports, nil
 }
 
+func (c *Controller) fetchSelectedSvc(namespace string, selector *metav1.LabelSelector) ([]string, error) {
+	sel, err := metav1.LabelSelectorAsSelector(selector)
+	if err != nil {
+		return nil, fmt.Errorf("error createing label selector, %v", err)
+	}
+	pods, err := c.podsLister.Pods(namespace).List(sel)
+	if err != nil {
+		return nil, fmt.Errorf("failed to list pods, %v", err)
+	}
+
+	svcIps := make([]string, 0)
+	for _, pod := range pods {
+		if !isPodAlive(pod) {
+			continue
+		}
+		if !pod.Spec.HostNetwork && pod.Annotations[util.AllocatedAnnotation] == "true" {
+
+			svcs, err := c.servicesLister.Services(pod.Namespace).List(labels.Everything())
+			if err != nil {
+				klog.Errorf("fetchSelectedSvc -> failed to list svc, %v", err)
+				return nil, err
+			}
```

Comment:
we can move this part to outside of the loop.

Reply:
ok